src/geo/ui/widgets/histogram/model.js

```diff
@@ -58,12 +58,26 @@ module.exports = WidgetModel.extend({
   },
 
   parse: function(data) {
+    var numberOfBins = data.bins_count;
+    var width = data.bin_width;
+    var nulls = data.nulls_count;
+    var start = data.bins_start;
 
-    this._data.reset(data.bins);
+    var buckets = new Array(numberOfBins);
+    _.each(data.bins, function(b) {
+      buckets[b.bin] = _.omit(b, 'bins_number', 'bin_width', 'nulls_count');
+    });
+
+    for (var i = 0; i < numberOfBins; i++) {
+      var binStart = start + (i * width);
+      var binEnd = start + ((i + 1) * width);
+      buckets[i] = _.extend({ bin: i, start: binStart, end: binEnd, freq: 0 }, buckets[i]);
```

Comment:
I've seen similar patterns here and there, I'd propose to change this code block to:
```js
_. defaults(buckets[i], {
  bin: i,
  start: start + (i * width),
  end: start + ((i + 1) * width),
  freq: 0
}
```

…due to:
1. Although a one-liner can be more concise it's actually harder to read and follow what's going on. IMHO one-liner objects should only be used when there's for 1-2 props max, e.g. `{simple:'enough'}`
2. binStart/binEnd variables aren't really necessary, especially if having the bucket definition on multiple lines it's clear what the calculations represent (compare above)
3. Use `_.defaults` to modify the `buckets[i]` directly instead of having to create a copy object and do an assignment. I assume start/end props not to exist on the object though so disregard this if that's isn't the case